### ISSUE TITLE
Route wildcards get confused with query params

### ISSUE THREAD

**Issue by karinehovhannisyan** [ThreeMammals/Ocelot#1452]
Hi, we lately faced a weird situation with wildcards and query params in Ocelot.

## Expected Behavior
We would expect the wildcards to have no relation to anything that comes after the **?** symbol and proceed the request as is.

## Actual Behavior
If a route query parameter contains the value of the wildcard, its first symbol is truncated and the parameter name is changed.

## Steps to Reproduce the Problem
Here is how we got this.

  1. Added a route like this (some fields are missing here)
  ```
{
      "DownstreamPathTemplate": "/api/{version}/Companies/{companyId}/all",
      "DownstreamScheme": "http",
      "DownstreamHostAndPorts": [
        {
          ...
        }
      ],
      "UpstreamPathTemplate": "/api/{version}/Companies/{companyId}/all",
      "UpstreamHttpMethod": [ "Get" ],
    }
```
  2. Make a request with this URL: ```http://localhost/api/v2/Companies/111/all?companyIds=111&companyIds=123```
  3. Our downstream receives the following params:
  
- ompanyIds - 111
- companyIds - 123

I also tried to add the ```{everything}``` after the **?**, but then if one makes a call like this:
```http://localhost/api/v2/Companies/111/all/?companyIds=111&companyIds=123```

They receive a 404 error.

## Specifications

  - Version: 17.0.0
  - Platform: .NET 5 application
  - Subsystem:

**Comment by YouvanigaTestsys** [comment]
Hi I am also facing the same , can anyone help me?

**Comment by raman-m** [comment]
It should be already fixed in v20.0
Are you interested in details?

**Comment by raman-m** [comment]
Hi @karinehovhannisyan!
This is well-known bug and it was fixed in v20.0
You need to re-define upstream template as following:
```json
{
    "DownstreamPathTemplate": "/api/{version}/Companies/{companyId}/all?{everything}",
    "UpstreamPathTemplate": "/api/{version}/Companies/{companyId}/all?{everything}",
}
```
See more here 👉 
- [Catch All Query String](https://ocelot.readthedocs.io/en/latest/features/routing.html#catch-all-query-string)

But...
Your both templates are identical! They are the same, no transformations, no changes at all.
I'm this case you can define one [Catch All](https://ocelot.readthedocs.io/en/latest/features/routing.html#id1) route:
```json
{
    "DownstreamPathTemplate": "/{everything}",
    "UpstreamPathTemplate": "/{everything}",
}
```
It should route any URL with any query strings.

**Comment by raman-m** [comment]
Already fixed by #1182
